__tests__/DMT.test.js

```diff
@@ -167,53 +188,57 @@ describe('Test suite for device mode transformation feature', () => {
     expect(payload.batch[0].event).toBe(event);
   });
 
-  it('Transformation server returning response in right format in case of successful transformation', () => {
+  it('Transformation server returning response in right format in case of successful transformation', async () => {
     window.XMLHttpRequest = jest.fn(() => xhrMockSuccess);
-    setTimeout(() => {
-      xhrMockSuccess.onreadystatechange();
-    }, 0);
-    return sendEventForTransformation(payload, 'write-key', 'data-plane-url', retryCount).then(
-      (response) => {
-        expect(xhrMockSuccess.send).toHaveBeenCalledTimes(1);
-        expect(Array.isArray(response.transformedPayload)).toEqual(true);
-        expect(typeof response.transformationServerAccess).toEqual('boolean');
-
-        const destObj = response.transformedPayload[0];
-
-        expect(typeof destObj).toEqual('object');
-        expect(destObj.hasOwnProperty('id')).toEqual(true);
-        expect(destObj.hasOwnProperty('payload')).toEqual(true);
-      },
-    );
+
+    await dmtHandler.sendEventForTransformation(payload, retryCount).then((response) => {
+      expect(typeof response.transformationServerAccess).toEqual('boolean');
+      expect(response.transformationServerAccess).toEqual(true);
```

Comment:
The second assertion overrides the first one.

Reply:
Addressed.